lock-api/src/test/java/com/palantir/lock/client/MultiClientCommitTimestampGetterTest.java

```diff
@@ -0,0 +1,251 @@
+        verify(cache, times(2)).processGetCommitTimestampsUpdate(any(), any());
+    }
+
+    @Test
+    public void doesNotUpdateCacheIfClientNotServed() {
+        Namespace alpha = Namespace.of("alpha" + UUID.randomUUID());
+        Namespace beta = Namespace.of("beta" + UUID.randomUUID());
+
+        BatchElement<NamespacedRequest, Long> requestForAlpha = batchElementForNamespace(alpha);
+        BatchElement<NamespacedRequest, Long> requestForBeta = batchElementForNamespace(beta);
+
+        List<BatchElement<NamespacedRequest, Long>> allRequests = ImmutableList.of(requestForAlpha, requestForBeta);
+        List<BatchElement<NamespacedRequest, Long>> alphaRequestList = ImmutableList.of(requestForAlpha);
+        Map<Namespace, GetCommitTimestampsResponse> responseMap = getCommitTimestamps(alphaRequestList);
+
+        when(timelockService.getCommitTimestamps(any())).thenReturn(responseMap).thenThrow(EXCEPTION);
+
+        assertThatThrownBy(() -> consumer.accept(allRequests)).isEqualTo(EXCEPTION);
+
+        // assert requests made by client alpha are served
+        assertSanityOfResponse(alphaRequestList, ImmutableMap.of(alpha, ImmutableList.of(responseMap.get(alpha))));
+
+        LockWatchEventCache alphaCache = LOCK_WATCH_EVENT_CACHE_MAP_MAP.get(alpha);
+        verify(alphaCache).lastKnownVersion();
+        verify(alphaCache).processGetCommitTimestampsUpdate(any(), any());
+
+        // assert unsuccessful requests were made by client beta
+        assertThat(requestForBeta.result().isDone()).isFalse();
+
+        LockWatchEventCache betaCache = LOCK_WATCH_EVENT_CACHE_MAP_MAP.get(beta);
+        verify(betaCache, times(2)).lastKnownVersion();
+        verify(betaCache, never()).processGetCommitTimestampsUpdate(any(), any());
+    }
+
+    private void setupServiceAndAssertSanityOfResponse(List<BatchElement<NamespacedRequest, Long>> batch) {
+        Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap = new HashMap<>();
+
+        when(timelockService.getCommitTimestamps(any())).thenAnswer(invocation -> {
+            Map<Namespace, GetCommitTimestampsResponse> commitTimestamps =
+                    getCommitTimestampResponse(invocation.getArgument(0));
+            commitTimestamps.forEach((namespace, response) -> {
+                expectedResponseMap
+                        .computeIfAbsent(namespace, _unused -> new ArrayList())
+                        .add(response);
+            });
+            return commitTimestamps;
+        });
+
+        consumer.accept(batch);
+        assertSanityOfResponse(batch, expectedResponseMap);
+    }
+
+    private void assertSanityOfResponse(
+            List<BatchElement<NamespacedRequest, Long>> batch,
+            Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap) {
+        // all requests should be served
+        assertThat(batch.stream().filter(elem -> !elem.result().isDone()).collect(Collectors.toSet()))
+                .isEmpty();
```

Comment:
It's usually worth having an `as(...)` clause as opposed to a comment here, to make errors more readable when a test fails